src/core/control/XournalMain.cpp

```diff
@@ -558,7 +557,8 @@ auto on_handle_local_options(GApplication*, GVariantDict*, XMPtr app_data) -> gi
     if (app_data->pdfFilename && app_data->optFilename && *app_data->optFilename) {
         return exec_guarded(
                 [&] {
-                    return exportPdf(*app_data->optFilename, app_data->pdfFilename, app_data->exportRange,
+                    return exportPdf(Util::fromGFilename(*app_data->optFilename),
+                                     Util::fromGFilename(app_data->pdfFilename), app_data->exportRange,
```

Comment:
> I may be confused here, but I think this is deleting app_data->pdfFilename. Is this wanted?? (there are similar cases below)

I need to recheck this, `Util::fromGFilename` is overloaded it only deletes when the char* is not const (it may be the case here), but in any way, that interface is weird and dangerous to use, we should rework it. My choice would be, to only accept a string_view. But then we would require to rewrite every call. Another idea is, to only accept Wrapped pointers. Then we would be required to change all calls, but at least, the compiler fails to compile, when they aren't wrapped.

Reply:
Introduced `Util::fromGFilenameUnchecked` to enforce correct handling.